src/ci/citool/src/cpu_usage.rs

```diff
@@ -0,0 +1,22 @@
+use std::path::Path;
+
+/// Loads CPU usage records from a CSV generated by the `src/ci/scripts/collect-cpu-stats.sh`
+/// script.
+pub fn load_cpu_usage(path: &Path) -> anyhow::Result<Vec<f64>> {
+    let reader = csv::ReaderBuilder::new().flexible(true).from_path(path)?;
+
+    let mut entries = vec![];
+    for row in reader.into_records() {
+        let row = row?;
+        let cols = row.into_iter().collect::<Vec<&str>>();
+
+        // The log might contain incomplete rows or some Python exception
+        if cols.len() == 2 {
+            if let Ok(idle) = cols[1].parse::<f64>() {
+                entries.push(100.0 - idle);
+            }
```

Comment:
shouldn't we at least emit a warning if we fail to parse the value?

Reply:
Not sure if anyone will ever read it :laughing: But why not.

Reply:
From this pr, logs contains `Warning: cannot parse CPU CSV entry Idle`, see https://github.com/rust-lang-ci/rust/actions/runs/13679120804/job/38246735471 or more recent https://github.com/rust-lang-ci/rust/actions/runs/14074774606/job/39415751601